### ISSUE TITLE
[BUG] Net 9 Android - StatusBar Color overriden on modal pages

### ISSUE THREAD

**Issue by Zack-G-I-T** [CommunityToolkit/Maui#2370]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [x] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

In MAUI Android, when adding StatusBarBehavior to change the color, and then going to a modal page - the status bar color gets replaced with the colorPrimary color from colors.xml.

This only happens on NET 9. In NET 8 the status bar color remained the same as the previous page.
I think its because int NET 9 in the App.cs you have to create a new Window for the app shell, but is there a solution to the status bar changing?

`protected override Window CreateWindow(IActivationState? activationState)
{
    var window = new Window(new AppShell());
    return window;
}`

### Expected Behavior

In the modal page, the status bar color should be the same as the previous normal page. This was the behaviour in NET 8

### Steps To Reproduce

1. Open repo
2. Run in NET 9 for Android and click on the button - see the status bar color change from orange to green/red.
3. Change to NET 8 - see the color stays orange throughout.

### Link to public reproduction project repository

https://github.com/Zack-G-I-T/StatusBarIssue-Net9

### Environment

```markdown
- .NET MAUI CommunityToolkit:
- OS:
- .NET MAUI:
```

**Comment by nixkuroi** [comment]
The Xaml override breaks the build as well.  I  had to comment out the following in MainPage.xaml for it to work.

<mct:StatusBarBehavior StatusBarColor="{Binding AppColor}"></mct:StatusBarBehavior>

**Comment by gabsamples6** [comment]
@brminnick  Hi , you linked a generic pull request , are you saying that #2215 will fix the issue? do you have a nightly build release that we can test? thanks again for  your free time you dedicate to this project

**Comment by pictos** [comment]
@gabsamples6 we aren't sure that will fix the this issue. But right now the MCT doesn't support .Net 9, it will when that PR gets merged, and if that doesn't fix this issue then we will investigate.

Probably the reason this is failing is because the change I made on how modal works in .Net Maiu, so we will need to update here, but any  .net9 fix is blocked by the linked PR

**Comment by gabsamples6** [comment]
@pictos thanks for your reply, loved your video on memory leaks! any idea when you will release .net 9 version as the .net maui team released the nuget SR2

**Comment by bijington** [comment]
I don't think they have release SR2 yet. I think it's due out tomorrow. Once it is out I expect us to work hard to get it done quickly

**Comment by bijington** [comment]
Sadly you are correct. They released SR1.4 but we need SR2

**Comment by gabsamples6** [comment]
@bijington  actually they seem to have updated the release notes and there way more stuff that original posted when they released the nuget. Does it have what you need to release the .net 9 version of the toolkit

**Comment by bijington** [comment]
@gabsamples6 it does have the fix but it also has brought in another issue that we are now having to try and diagnose.

**Comment by Zack-G-I-T** [comment]
@brminnick @bijington @gabsamples6  I tested the linked repo using the latest package references and the issue still occurs. Should this actually be marked as completed?

	<PackageReference Include="CommunityToolkit.Maui" Version="10.0.0" />
	<PackageReference Include="CommunityToolkit.Mvvm" Version="8.4.0" />
	<PackageReference Include="Microsoft.Maui.Controls" Version="9.0.21" />

**Comment by gabsamples6** [comment]
Hi just to say that this is not fixed ticket should be re-opened

**Comment by pictos** [comment]
This is something expected since .net maui changed the way how modal works, now it uses a DialogFragment, and it has it own Window, and that Window doesn't apply the same configuration as the Activity.Window. I'll need to think on how to re-design the code for StatusBar color, right now the work around should be this one
<details>


```cs
#if ANDROID
using Android.App;
using AndroidX.Activity;
using AndroidX.Fragment.App;
using FragmentManager = AndroidX.Fragment.App.FragmentManager;
using Microsoft.Maui.Platform;
using Android.Views;
using AndroidX.Core.View;
#endif
using CommunityToolkit.Maui;
using Microsoft.Extensions.Logging;
using Microsoft.Maui.LifecycleEvents;
using System.Diagnostics;

namespace TestStatusBar;

public static class MauiProgram
{
	public static MauiApp CreateMauiApp()
	{
		var builder = MauiApp.CreateBuilder();
		builder
			.UseMauiApp<App>()
			.UseMauiCommunityToolkit()
			.ConfigureFonts(fonts =>
			{
				fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
				fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
			})
			.ConfigureLifecycleEvents(a =>
			{
#if ANDROID
				a.AddAndroid(builder =>
				{
					builder.OnCreate((activity, bundle) =>
					{
						if (activity is not ComponentActivity componentActivity)
						{
							return;
						}
						componentActivity.GetFragmentManager()?.RegisterFragmentLifecycleCallbacks(new MyFragmentLifecycleCallbacks((fm, f) =>
						{
							if (f is AndroidX.Fragment.App.DialogFragment dialogFragment)
							{
								var activity = Platform.CurrentActivity;

								if (activity is null)
									return;

								var statusBarColor = activity.Window!.StatusBarColor;
								var platformColor = new Android.Graphics.Color(statusBarColor);

								var dialog = dialogFragment.Dialog;

								Debug.Assert(dialog is not null);
								Debug.Assert(dialog.Window is not null);

								var window = dialog.Window;
								// Edge to Edge on the fragment's window
								//var platformColor = color.ToPlatform();
								dialog.Window.SetStatusBarColor(platformColor);

								bool isColorTransparent = platformColor == Android.Graphics.Color.Transparent;
								if (isColorTransparent)
								{
									window.ClearFlags(WindowManagerFlags.DrawsSystemBarBackgrounds);
									window.SetFlags(WindowManagerFlags.LayoutNoLimits, WindowManagerFlags.LayoutNoLimits);
								}
								else
								{
									window.ClearFlags(WindowManagerFlags.LayoutNoLimits);
									window.SetFlags(WindowManagerFlags.DrawsSystemBarBackgrounds, WindowManagerFlags.DrawsSystemBarBackgrounds);
								}

								window.SetDecorFitsSystemWindows(!isColorTransparent);


								CommunityToolkit.Maui.Core.Platform.StatusBar.SetColor(Color.FromArgb("#FFA500"));
							}

						}), false);
					});
				});
#endif
			});
		Routing.RegisterRoute("SecondPage", typeof(SecondPage));
		builder.Services.AddTransient<SecondPage>();
#if DEBUG
		builder.Logging.AddDebug();
#endif

		return builder.Build();
	}
}


#if ANDROID
public class MyFragmentLifecycleCallbacks(Action<AndroidX.Fragment.App.FragmentManager, AndroidX.Fragment.App.Fragment> onFragmentStarted) : FragmentManager.FragmentLifecycleCallbacks
{
	public override void OnFragmentStarted(FragmentManager fm, AndroidX.Fragment.App.Fragment f)
	{
		onFragmentStarted?.Invoke(fm, f);
		base.OnFragmentStarted(fm, f);
	}
}
#
```` 


</details>

**Comment by pictos** [comment]
@WMLPB didn't touch on navigationBar, if you can provide a small project that reproduces the issue I can take a look and do my best to include it on my current PR

**Comment by pictos** [comment]
@WMLPB 

> But, thinking about it… – isn’t this actually a bug in MAUI itself? Shouldn’t the DialogFragment use the same settings like the active Activity in the first place? Shouldn’t the MAUI team fix this?

It's more an Android thing. On Maui we changed the modal navigation to use DialogFragment in order to make it more _natural_ , but the DialogFragment has it own Window and that Window doesn't inherit the properties values and behavior of the MainActivity.Window (At least we didn't find a way to do it), so when a modal appears now this happens and would be our responsibility to make adjust that values. Probably my PR here will go to Maui, that way all 3rd vendors and devs will have a single place to configure and customize their modals
